From 5f9087117e4e7d40e6da492275b435bbcdab23e5 Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Sun, 24 Jun 2018 22:59:34 +0200
Subject: [PATCH 3/3] Backported and rebased patch from Debian maintainer
 Nicholas Guriev. Removed links to packaged Open Sans fonts. Telegram Desktop
 will use packaged versions.

---
 Telegram/Resources/qrc/telegram.qrc | 3 ---
 Telegram/SourceFiles/ui/twidget.cpp | 6 +++---
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/Telegram/Resources/qrc/telegram.qrc b/Telegram/Resources/qrc/telegram.qrc
index 91cbeee5c..fbffd8abb 100644
--- a/Telegram/Resources/qrc/telegram.qrc
+++ b/Telegram/Resources/qrc/telegram.qrc
@@ -42,9 +42,6 @@
     <file alias="js/script.js">../export_html/js/script.js</file>
   </qresource>
   <qresource prefix="/gui">
-    <file alias="fonts/OpenSans-Regular.ttf">../fonts/OpenSans-Regular.ttf</file>
-    <file alias="fonts/OpenSans-Bold.ttf">../fonts/OpenSans-Bold.ttf</file>
-    <file alias="fonts/OpenSans-Semibold.ttf">../fonts/OpenSans-Semibold.ttf</file>
     <file alias="art/bg.jpg">../art/bg.jpg</file>
     <file alias="art/bg_initial.jpg">../art/bg_initial.jpg</file>
     <file alias="art/logo_256.png">../art/logo_256.png</file>
diff --git a/Telegram/SourceFiles/ui/twidget.cpp b/Telegram/SourceFiles/ui/twidget.cpp
index 71f318229..8912791a3 100644
--- a/Telegram/SourceFiles/ui/twidget.cpp
+++ b/Telegram/SourceFiles/ui/twidget.cpp
@@ -70,9 +70,9 @@ void Start() {
 	}
 	Started = true;
 
-	auto regular = LoadCustomFont(qsl(":/gui/fonts/OpenSans-Regular.ttf"), qsl("Open Sans"));
-	auto bold = LoadCustomFont(qsl(":/gui/fonts/OpenSans-Bold.ttf"), qsl("Open Sans"), style::internal::FontBold);
-	auto semibold = LoadCustomFont(qsl(":/gui/fonts/OpenSans-Semibold.ttf"), qsl("Open Sans Semibold"));
+	auto regular = ValidateFont(qsl("Open Sans"));
+	auto bold = ValidateFont(qsl("Open Sans"), style::internal::FontBold);
+	auto semibold = ValidateFont(qsl("Open Sans Semibold"));
 
 #ifdef Q_OS_WIN
 	// Attempt to workaround a strange font bug with Open Sans Semibold not loading.
-- 
2.17.1

